cognite/neat/rules/models/_rules/_types/_base.py

```diff
@@ -0,0 +1,228 @@
+
+    type_: ClassVar[EntityTypes] = EntityTypes.undefined
+    prefix: str | Undefined = Undefined
+    suffix: str
+    version: str | None = None
+    name: str | None = None
+    description: str | None = None
+
+    def __lt__(self, other: object) -> bool:
+        if type(self) is not type(other) or not isinstance(other, Entity):
+            return NotImplemented
+        return self.versioned_id < other.versioned_id
+
+    def __eq__(self, other: object) -> bool:
+        if type(self) is not type(other) or not isinstance(other, Entity):
+            return NotImplemented
+        return self.versioned_id == other.versioned_id
+
+    def __hash__(self) -> int:
+        return hash(self.versioned_id)
+
+    @property
+    def id(self) -> str:
+        if self.prefix is Undefined:
+            return self.suffix
+        else:
+            return f"{self.prefix}:{self.suffix}"
+
+    @property
+    def versioned_id(self) -> str:
+        if self.version is None:
+            return self.id
+        else:
+            return f"{self.id}(version={self.version})"
+
+    @property
+    def space(self) -> str:
+        """Returns entity space in CDF."""
+        return self.prefix
+
+    @property
+    def external_id(self) -> str:
+        """Returns entity external id in CDF."""
+        return self.suffix
+
+    def __repr__(self):
+        return self.versioned_id
+
+    def __str__(self):
+        return self.versioned_id
+
+    @classmethod
+    def from_string(cls, entity_string: str, base_prefix: str | None = None) -> Self:
+        if result := VERSIONED_ENTITY_REGEX_COMPILED.match(entity_string):
+            return cls(
+                prefix=result.group("prefix"),
+                suffix=result.group("suffix"),
+                version=result.group("version"),
+            )
+        elif result := ENTITY_ID_REGEX_COMPILED.match(entity_string):
+            return cls(prefix=result.group("prefix"), suffix=result.group("suffix"))
+        elif base_prefix and re.match(SUFFIX_REGEX, entity_string) and re.match(PREFIX_REGEX, base_prefix):
+            return cls(prefix=base_prefix, suffix=entity_string)
+        else:
+            raise ValueError(f"{cls.__name__} is expected to be prefix:suffix, got {entity_string}")
+
+    @classmethod
+    def from_list(cls, entity_strings: list[str], base_prefix: str | None = None) -> list[Self]:
+        return [
+            cls.from_string(entity_string=entity_string, base_prefix=base_prefix) for entity_string in entity_strings
+        ]
+
+
+class ContainerEntity(Entity):
+    type_: ClassVar[EntityTypes] = EntityTypes.container
+
+    @classmethod
+    def from_raw(cls, value: Any) -> "ContainerEntity":
+        if not value:
+            return ContainerEntity(prefix=Undefined, suffix=value)
+        elif isinstance(value, ContainerEntity):
+            return value
+
+        if ENTITY_ID_REGEX_COMPILED.match(value) or VERSIONED_ENTITY_REGEX_COMPILED.match(value):
```

Comment:
Containers does not have a version. Alternative is to raise an exception if it matches the versioned entity regex 
```suggestion
        if ENTITY_ID_REGEX_COMPILED.match(value):
```